src/AaveV2CollectorContractConsolidation.sol

```diff
@@ -0,0 +1,258 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.17;
+
+import {AaveV2Ethereum} from "@aave-address-book/AaveV2Ethereum.sol";
+import {AggregatorV3Interface} from "./external/AggregatorV3Interface.sol";
+import {ERC20} from "@openzeppelin/token/ERC20/ERC20.sol";
+import {SafeERC20} from "@openzeppelin/token/ERC20/utils/SafeERC20.sol";
+
+/// @title AaveV2CollectorContractConsolidation
+/// @author Llama
+/// @notice Contract to swap excess assets for USDC at a discount
+contract AaveV2CollectorContractConsolidation {
```

Comment:
You might need to add a rescue function as well for tokens that are accidentally sent directly to this contract. I'm pretty sure BGD is going to ask for that. You can take a look at the 100K BAL Bonding curve contract on how to do that: https://github.com/llama-community/aave-bal-bonding-curve/blob/main/src/OneWayBondingCurve.sol#L116-L122

And ofc, add appropriate test cases for that.